File: ARCHIVOS JAVA/pruebaArchivos.java
package Archivos;

import java.io.*;
import java.util.ArrayList;

public class pruebaArchivos {
    private ArrayList<String> contenidoArchivo = new ArrayList<>();
    private ArrayList<JornadaLaboral> listaEntrada = new ArrayList<>();
    private final String rutaWindows = "D:\\TECNOLOGICO\\SEMESTRE 2\\POO\\JornadaLaboral\\Archivos\\Datos.txt";

    public ArrayList<String> getContenidoArchivo() {
        return contenidoArchivo;
    }

    public ArrayList<JornadaLaboral> getListaEntrada() {
        return listaEntrada;
    }

    public void leerArchivo(){
        LeerArchivo leer = new LeerArchivo();
        this.contenidoArchivo =
                leer.leerArchivo(this.rutaWindows);
        for (int i=0; i < this.contenidoArchivo.size();i++){
            String linea = this.contenidoArchivo.get(i);
            String elementosLinea[] = linea.split("#");
            try {
                int horasTrabajadas = Integer.parseInt(elementosLinea[0]);
                int horasExtras = Integer.parseInt(elementosLinea[1]);
                double salario = Double.parseDouble(elementosLinea[2]);
                listaEntrada.add(new JornadaLaboral(horasTrabajadas, horasExtras, salario));
            } catch (NumberFormatException ex){
                ex.printStackTrace();
                listaEntrada.add(new JornadaLaboral(0, 0, 0.0));
            }
        }
    }

    public void escribirArchivo(ArrayList<JornadaLaboral> lista){
        String archivo = "resultados.csv";
        File f = new File(archivo);
        //Escritura
        try{
            FileWriter w = new FileWriter(f);
            BufferedWriter bw = new BufferedWriter(w);
            PrintWriter wr = new PrintWriter(bw);
            wr.write("");
            wr.append("Horas trabajadas, horas extra, salario, total\n");
            for (int i=0; i < lista.size(); i++){
                wr.append(lista.get(i).getHorasTrabajadas() +
                        "\t \t" + lista.get(i).getHorasExtras() +
                        "\t \t" + lista.get(i).getSalario() +
                        "\t \t" + lista.get(i).getSalarioTotal() + "\n");
            }
            wr.close();
            bw.close();
        }catch(IOException e){

        }
    }

    public static void main (String args[]){
        pruebaArchivos prueba = new pruebaArchivos();
        prueba.leerArchivo();
        for (int i=0; i < prueba.getListaEntrada().size(); i++){
            System.out.println(prueba.getListaEntrada().get(i).getHorasTrabajadas() +
                    "\t" + prueba.getListaEntrada().get(i).getHorasExtras() +
                    "\t" + prueba.getListaEntrada().get(i).getSalario() + "\t" +
                    prueba.getListaEntrada().get(i).getSalarioTotal());
        }
        prueba.escribirArchivo(prueba.getListaEntrada());

    }

}

File: ARCHIVOS PYTHON/PruebaArchivos.py
from abc import ABCMeta

class JornadaLaboral(ABCMeta):
    #Método de clase
    @staticmethod
    def calcularSalarioNormal(salario,horasTrabajadas):
        total = salario * horasTrabajadas
        return total

class pruebaArchivos:

    def leerArchivo(self, archivo):
        file = open(archivo, 'r')
        lineas = []
        lineas_archivo = []
        for linea in file.readlines():
            lineas.append(linea.replace('\n','').split(",")) #nos permite hacer el arreglo multi o el recorrido
        file.close()

        for f in range(0, len(lineas)):
            try:
                lineas_archivo.append([float(lineas[f][0]), int (lineas[f][1])])
            except ValueError:
                lineas_archivo.append(([0,0]))
        return lineas_archivo

    def Empleado (self, lista):
        resultados = []
        for f in range (0, len(lista)):
            resultados.append(JornadaLaboral.calcularSalarioNormal(
                lista[f][0], lista[f][1]))
        return resultados

    def guardarResultados(self, entrada, resultados):
        file = open("salario.csv", 'w')
        file.write('salario,horasTrabajadas,total\n')
        for f in range(0, len(entrada)):
            linea = str(entrada[f][0]) + '\t \t' + \
                    str(entrada[f][1]) + '\t \t' + \
                    str(resultados[f]) + '\n'
            file.write(linea)
        file.close()

if __name__ == "__main__":
      prueba = pruebaArchivos()
      archivo = []
      archivo = prueba.leerArchivo("Datos.txt")
      resultado = prueba.Empleado(archivo)
      prueba.guardarResultados(archivo,resultado)
